cpp/modmesh/inout/plot3d.cpp

```diff
@@ -0,0 +1,148 @@
+#include <modmesh/inout/plot3d.hpp>
+
+namespace modmesh
+{
+namespace inout
+{
+Plot3d::Plot3d(const std::string & data)
+{
+    std::string line;
+    std::queue<real_type> parsing_q;
+    size_t total_blk_size = 0;
+    size_t x_base_idx = 0, y_base_idx = 0, z_base_idx = 0;
+    uint_type idx = 0;
+    uint_type nblocks = 0;
+    uint_type base_idx = 0;
+    // hexahedron has 8 nodes + 1 for store nodes number
+    small_vector<uint_type> nds_temp(9, 0U);
+    // These vector of x, y, z coordinate shift is used to build the hexahedron element
+    // the nodes order of the hexahedron element is clock of the xy face
+    // Referece: https://github.com/nasa/Plot3D_utilities/blob/main/colab/Plot3D_SplitBlocksExample.ipynb
+    small_vector<small_vector<int_type>> shift = {
```

Comment:
`std::vector<std::vector<int_type>>` is good enough because there are only initial memory allocation.

You can also try `constexpr` if the contents are not changed later.